org.eclipse.xtext.tests/src/org/eclipse/xtext/resource/XtextResourceTest.java

```diff
@@ -141,4 +141,12 @@ public void unloadRoot(EObject root) {
 		assertEquals(1, diag.getLength());
 	}
 
+	@Test public void testClearCache() throws Exception {
+		resource.update(0, 0, simpleModel);
+		Object cv1 = resource.getCache().get("justatest", resource, ()->new Object());
+		resource.update(0, 0, "");
+		Object cv2 = resource.getCache().get("justatest", resource, ()->new Object());
+		assertFalse(cv1 == cv2);
```

Comment:
just cosmetic: use `assertNotSame` instead of `assertFalse` with a boolean condition